src/main/cc/proxy_client/proxy_client.cc

```diff
@@ -517,8 +520,15 @@ int CreateRunRequest(int argc, char** argv, const char** env,
     string varval(*env++);
     unsigned int eq_index = varval.find("=");
     string var = varval.substr(0, eq_index);
-    if (var == "PWD")
+    if (var == "PWD") {
       absl::StrAppend(&env_vars, varval, ",");
+      continue;
+    }
+    for (const auto& whitelisted_var : absl::StrSplit(FLAGS_env_whitelist, ',', absl::SkipEmpty())) {
```

Comment:
Don't do that every time, read this into a set once, and then do set lookups for every variable. Also, just add PWD to the set to begin with.

Reply:
Done.